pkg/3scale/amp/operator/upgrade.go

```diff
@@ -758,6 +767,103 @@ func (u *UpgradeApiManager) findDeploymentTriggerOnImageChange(triggerPolicies [
 	return result, nil
 }
 
+func (u *UpgradeApiManager) upgradePodTemplateLabels() (reconcile.Result, error) {
+	apicast, err := Apicast(u.apiManager, u.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	backend, err := Backend(u.apiManager, u.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	zync, err := Zync(u.apiManager, u.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	memcached, err := Memcached(u.apiManager)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	system, err := System(u.apiManager, u.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	deploymentConfigs := []*appsv1.DeploymentConfig{
+		apicast.StagingDeploymentConfig(),
+		apicast.ProductionDeploymentConfig(),
+		backend.ListenerDeploymentConfig(),
+		backend.WorkerDeploymentConfig(),
+		backend.CronDeploymentConfig(),
+		zync.DeploymentConfig(),
+		zync.QueDeploymentConfig(),
+		zync.DatabaseDeploymentConfig(),
```

Comment:
This should depend on whether zync-database is set as an external database or not.

Reply:
good catch. I copy&pasted the upgrade procedure when zync database could not be externalized.

Reply:
fixed